optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/acceptor/greatdeluge/GreatDelugeAcceptor.java

```diff
@@ -0,0 +1,118 @@
+    private double[] levelScoreLevels;
+
+    private double levelMinimum = 0;
+    private static final double THRESHOLD = .0000001;
+
+    // Good value to come out from. Source: https://github.com/UniTime/cpsolver from Tomas Muller
+    private static final double DEFAULT_RAIN_SPEED_RATIO = 0.99_999_995;
+
+    public void setInitialLevels(Score initialLevel) { this.initialLevel = initialLevel; }
+
+    public Score getInitialLevels() { return this.initialLevel; }
+
+    public void setRainSpeed(Double rainSpeed) {
+        this.rainSpeed = rainSpeed;
+        this.rainSpeedRatio = null;
+    }
+
+    public Double getRainSpeed() { return this.rainSpeed; }
+
+    public void setRainSpeedRatio(Double rainSpeedRatio) {this.rainSpeedRatio = rainSpeedRatio; }
+
+    public Double getRainSpeedRatio() { return this.rainSpeedRatio; }
+
+    public double[] getLevelScoreLevels() { return this.levelScoreLevels; }
+
+    public void phaseStarted(LocalSearchPhaseScope phaseScope) {
+        super.phaseStarted(phaseScope);
+
+        if (initialLevel != null) {
+            for (double initialLevelLevel : ScoreUtils.extractLevelDoubles(initialLevel)) {
+                if (initialLevelLevel < 0.0) {
+                    throw new IllegalArgumentException("The initial level (" + initialLevel
+                            + ") cannot have negative level (" + initialLevelLevel + ").");
+                }
+            }
+            initialLevelScoreLevels = ScoreUtils.extractLevelDoubles(initialLevel);
+            levelScoreLevels = initialLevelScoreLevels;
+        } else {
+            double[] initialLevelScoreLevelsNegative;
+            initialLevelScoreLevelsNegative = ScoreUtils.extractLevelDoubles(phaseScope.getBestScore());
+            levelScoreLevels = initialLevelScoreLevelsNegative;
+
+            for (int i = 0; i < levelScoreLevels.length; i++) {
+                if (Math.abs(levelScoreLevels[i]) < THRESHOLD) {
+                    continue;
+                }
+                levelScoreLevels[i] = -levelScoreLevels[i];
+            }
+        }
+        levelsLength = levelScoreLevels.length;
+    }
+
+    public void phaseEnded(LocalSearchPhaseScope phaseScope) {
+        super.phaseEnded(phaseScope);
+        initialLevelScoreLevels = null;
+        levelScoreLevels = null;
+        levelsLength = -1;
+    }
+
+    @Override
+    public boolean isAccepted(LocalSearchMoveScope moveScope) {
+
+        Score moveScore = moveScope.getScore();
+        double[] moveScoreLevels = ScoreUtils.extractLevelDoubles(moveScore);
+
+        for (int i = 0; i < levelsLength; i++) {
+            double moveScoreLevel = moveScoreLevels[i];
+            double levelScoreLevel = levelScoreLevels[i];
+
+            double isZeroIfEqual = Math.abs(moveScoreLevel + levelScoreLevel);
+
+            if (moveScoreLevel > -levelScoreLevel) {
+                // Checks if given score is better than water level, if so, move is accepted.
```

Comment:
This comment does not add any extra information that the if statement on line 90 already does.